charts/opensearch/templates/statefulset.yaml

```diff
@@ -232,6 +232,17 @@ spec:
           - name: "{{ template "opensearch.uname" . }}"
             mountPath: {{ .Values.opensearchHome }}/data
 {{- end }}
+{{- if .Values.sysctlInitContainer }}
+      - name: sysctl
+        image: "{{ template "opensearch.dockerRegistry" . }}{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+        imagePullPolicy: "{{ .Values.image.pullPolicy }}"
+        command: ["sysctl", "-w", "vm.max_map_count={{ .Values.sysctlVmMaxMapCount}}"]
```

Comment:
I don't think we should write this every time the init container runs unless it is necessary. Maybe some logic like this to check the value and only write it if required? After all, you don't want to _reduce_ this value if some other service needs it to be even higher.
```
  command:
  - sh
  - -c
  - |
    set -xe
    DESIRED="{{ .Values.sysctl.VmMaxMapCount }}"
    CURRENT=$(sysctl -n vm.max_map_count)
    if [ "$DESIRED" -gt "$CURRENT" ]; then
      sysctl -w vm.max_map_count=$DESIRED
    fi
```

Reply:
Yup this will be un-necessary to do it. To write if only required is a good strategy

Reply:
Updated.